packages/shell-api/src/aggregation-cursor.ts

```diff
@@ -69,19 +69,25 @@ export default class AggregationCursor extends ShellApiClass {
     return this._cursor.isClosed();
   }
 
-  isExhausted(): Promise<boolean> {
-    return this._cursor.isExhausted();
+  async isExhausted(): Promise<boolean> {
+    return this._cursor.isClosed() && !await this._cursor.hasNext();
```

Comment:
`isExhausted` does not exist on the underlying server, so I assumed doing this like we do it for regular shell `Cusor` objects should be fine